dp/cloud/python/magma/radio_controller/services/active_mode_controller/service.py

```diff
@@ -146,6 +198,26 @@ def _update_cbsd(self, db_id: int, to_update: Dict) -> DBCbsd:
         return updated
 
 
+def _store_requests_from_map_in_db(session: Session, request_types_map: Dict[str, int], request_map: Dict[str, List[Dict]]) -> None:
+    request_type = next(iter(request_map))
+    for request_json in request_map[request_type]:
+        filters = get_cbsd_filter_strategies[request_type](request_json)
+        cbsd = session.query(DBCbsd).filter(*filters).first()
```

Comment:
instead of fetching the whole cbsd you could make this into a subquery that only fetches the id

Reply:
We can filter only id, but it cannot be a subquery, since we have to check if radio exists (for example there may be a scenario, were user updates serial number, which active mode controller is generating requests)